hcl2template/testdata/hcp_par/complete.pkr.hcl

```
@@ -0,0 +1,27 @@
+// starts resources to provision them.
+build {
+    name = "bucket-slug"
+
+    hcp_packer_registry {
+        description = <<EOT
+Some description
+    EOT
+        labels = {
+            "foo" = "bar"
+        }
+    }
```

I think that the bucket should specific and not be taken from the build block, this could get quite confusing ! Like what if I want to name my build with something but push to another bucket ?

```suggestion
    name = "build-name"

    hcp_packer_registry {
         push_to = "alpine"
        description = <<EOT
Some description
    EOT
        labels = {
            "foo" = "bar"
        }
    }
```

---

Another question on the same topic, why not taking the description from the build and put that in the bucket too ?

---

> I think that the bucket setting should be specific and not be taken from the build block, this could get quite confusing ! Like what if I want to name my build with something but push to another bucket ?

This is fair. From the minimal viable configuration option stand point " A user should be able to push to PAR simply by adding a few environment variables" so this keeps to this promise as long as a build name is part of the configuration template. Or the user provides a `HCP_PACKER_BUILD_NAME` env variable along side the HCP creds. 

That said I think that supporting a different build name in the future (it would be great to get beta feedback here) can be a simple thing to do. But for now I feels a little out of scope. Of course happy to be over ruled here.

---

Nice! I will make the same logic for both slug and description. If not present in hcp_packer_registry, then use the ones from build. 

Also, the description is not documented in the deployed docs. Is it missing or is it something new?

---

I'd think the opposite, I think it would be quite easier to make of that a variable, and then pass it to the build and hcp blocks, to not repeat myself.

---

> A user should be able to push to PAR simply by adding a few environment variables

To me, this is marketing, not a technical spec 😛

---

I'm adding the Slug option because it won't change the current behavior. It will just add the possibility of having a different name from the build name. The build name will still be used if no slug is provided, and the env variable will override whatever slug is set from the config.

---

Question: For the source references, what becomes the value of compnentType for some build,  it doesn't currently include the build name. So `learn-packer.docker.example` is stored as `docker.example`. Should it include the build name as well if set?

---

pinging @SwampDragons as well for thoughts on last question.

---

Adding a note here that I initially expected component type to be just the component that build that thing; so the `docker` builder, the `amazon-ebs` builder or the `google-import` pp; for example.

---

> To me, this is marketing, not a technical spec 😛

This is a technical spec. Otherwise it's not marketing, it's a lie.

---

When writing the service, I imagined "component-type" as literally the builder or post-processor that created the builder, without either name

---

Ah, yes, what Adrien said.